### PR TITLE
Implemented support for ServiceChecks

### PR DESCRIPTION
**Pull request by jovanbrakus**
Support for ServiceChecks is implemented based on DataDog wire protocol specification.

### CODE REVIEW

**Comment by truthbk** [comment]
@jovanbrakus sorry for the slow response, I realize you sbumitted this ages ago! We'll review this very shortly, thanks a lot for your patience! :+1:

**Comment by seiffert** [comment]
I stumbled upon the fact that event texts as well as service check messages can contain `\n` characters which need to be escaped (`\\n`). I just submitted a PR (#8) for the event texts and wanted to notify you about this. I haven't used service checks yet, but from the python client library, I can tell that the messages need to be escaped, too.

See https://github.com/DataDog/datadogpy/blob/b5f7d6d2748fad61b2e060768ce68e3b76d2c928/datadog/dogstatsd/base.py#L259-L263

**Comment by jovanbrakus** [comment]
Yup, agreed.
Will do that soon.

**Comment by truthbk** [comment]
Just a quick note, #8 has now been merged.

**Comment by jovanbrakus** [comment]
I added escaping of '\n' and 'm:' along with tests that show functionality...

**Comment by truthbk** [comment]
@jovanbrakus could you please rebase to the latest master?

File: statsd/statsd_test.go [hunk 1]
```diff
@@ -266,6 +266,76 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestServiceChecks(t *testing.T) {
```
Comment (truthbk) [inline statsd/statsd_test.go]: Thank you so much for writing this! :heart:

File: statsd/statsd_test.go [hunk 2]
```diff
@@ -266,6 +266,76 @@ func TestEvents(t *testing.T) {
...
+		}, {
+			&ServiceCheck{Name: "DataCatService", Status: Ok, Hostname: "DataStation.Cat", Message: "Here goes valuable message", Tags: []string{"host:foo", "app:bar"}},
+			`_sc|DataCatService|0|h:DataStation.Cat|#host:foo,app:bar|m:Here goes valuable message`,
+		}, {
+			&ServiceCheck{Name: "DataCatService", Status: Ok, Hostname: "DataStation.Cat", Message: "Here goes \n that should be escaped", Tags: []string{"host:foo", "app:bar"}},
+			`_sc|DataCatService|0|h:DataStation.Cat|#host:foo,app:bar|m:Here goes \n that should be escaped`,
+		}, {
+			&ServiceCheck{Name: "DataCatService", Status: Ok, Hostname: "DataStation.Cat", Message: "Here goes m: that should be escaped", Tags: []string{"host:foo", "app:bar"}},
+			`_sc|DataCatService|0|h:DataStation.Cat|#host:foo,app:bar|m:Here goes m\: that should be escaped`,
+		},
```
Comment (truthbk) [inline statsd/statsd_test.go]: Can you please add another case with some unicode in the message... should be fine, but it's worth checking. Thanks!

**Comment by truthbk** [comment]
Looks solid, if you could please add the unicode message test, rebase and squash, we should then be good to merge!

Thanks so much for your contrib!

**Comment by jovanbrakus** [comment]
Added unicode test and rebased it to latest master.

**Comment by truthbk** [comment]
@jovanbrakus I think you accidentally rebased on the wrong branch or branched off the wrong one, so we've got a few conflicts... Anyways I created a branch (jaime/svchk_support @ https://github.com/DataDog/datadog-go/compare/jaime/svchk_support) with your commits and no conflicts, if you want to `git reset --hard origin jaime/svchk_support`) and give it a last pass?

**Comment by jovanbrakus** [comment]
Haven't seen your msg before.... Branch svchk_support looks fine.

**Comment by diasjorge** [comment]
hi guys any chance this might get merged soon? we'd really like to use this functionality. Please :)

**Comment by jovanbrakus** [comment]
We're using ServiceChecks in production for months...

**Comment by jovanbrakus** [comment]
Closing this as it's been merged on another PR

**Pull request closed without merging**